### PR TITLE
Catch error and return 404 for module_fails with non existing test

### PR DESCRIPTION
**Pull request by asdil12**
Ticket: https://progress.opensuse.org/issues/121441

### CODE REVIEW

**Review by perlpunk — COMMENTED** [review]
++ but maybe add a test? There's already a /modules/.../fail test in t/10-tests_overview.t

File: lib/OpenQA/WebAPI/Controller/Test.pm [hunk 1]
```diff
@@ -727,6 +727,7 @@ sub module_fails {
             job_id => $self->param('testid'),
             name => $self->param('moduleid'),
         })->first;
+    return $self->reply->not_found unless defined $module;
```
Comment (kalikiana) [inline lib/OpenQA/WebAPI/Controller/Test.pm:730]: I'm with @perlpunk here. I think this would be good to cover in a unit test

**Review by Martchus — APPROVED** [review]
A test would be nice of course.

File: lib/OpenQA/WebAPI/Controller/Test.pm (hunk 1 above)
Comment (asdil12) [reply lib/OpenQA/WebAPI/Controller/Test.pm:730]: added test

**Review by Martchus — APPROVED** [review]

**Pull request merged**